src/ServiceControl.Infrastructure/PluginAssemblyLoadContext.cs

```diff
@@ -0,0 +1,41 @@
+﻿namespace ServiceControl.Infrastructure;
+
+using System.Reflection;
+using System.Runtime.Loader;
+
+public class PluginAssemblyLoadContext(string assemblyPath) : AssemblyLoadContext(assemblyPath)
+{
+    readonly AssemblyDependencyResolver resolver = new(assemblyPath);
+
+    protected override Assembly Load(AssemblyName assemblyName)
+    {
+        foreach (var assembly in Default.Assemblies)
+        {
+            if (assembly.FullName == assemblyName.FullName)
+            {
+                return null;
+            }
+        }
```

Comment:
```suggestion
        if (Default.Assemblies.Any(assembly => assembly.FullName == assemblyName.FullName)) return null;
```

Reply:
my assumption from reading this code is that if the assembly full name is in Defauly.Assemblies then it's already loaded and therefore doesn't need to be loaded here? Perhaps provide a comment to this effect